packages/main/src/plugin/extension/extension-development-folders.ts

```diff
@@ -0,0 +1,150 @@
+      this.#developmentFolders = updatedFolders;
+      this.#onDidUpdateDevelopmentFolders.fire(this.getDevelopmentFolders());
+    }
+  }
+
+  init(): void {
+    const developmentExtensionsFoldersConfiguration: IConfigurationNode = {
+      id: 'preferences.extensions',
+      title: 'Extensions',
+      type: 'object',
+      properties: {
+        [`${ExtensionDevelopmentFolderInfoSettings.SectionName}.${ExtensionDevelopmentFolderInfoSettings.DevelopmentExtensionsFolders}`]:
+          {
+            description: 'List of extension folders to monitor',
+            type: 'array',
+            default: [],
+            hidden: true,
+          },
+      },
+    };
+    this.#configurationRegistry.registerConfigurations([developmentExtensionsFoldersConfiguration]);
+
+    // refresh the value when the property is changed
+    this.#configurationRegistry.onDidChangeConfiguration(event => {
+      if (
+        event.key ===
+        `${ExtensionDevelopmentFolderInfoSettings.SectionName}.${ExtensionDevelopmentFolderInfoSettings.DevelopmentExtensionsFolders}`
+      ) {
+        this.refreshFolders();
+      }
+    });
+
+    // initialize and track all development folders
+    this.refreshFolders();
+
+    this.#firstLaunch = false;
+  }
+
+  protected async saveToConfiguration(): Promise<void> {
+    await this.#configurationRegistry
+      .getConfiguration(ExtensionDevelopmentFolderInfoSettings.SectionName)
+      .update(ExtensionDevelopmentFolderInfoSettings.DevelopmentExtensionsFolders, this.#developmentFolders);
+
+    // send an event to refresh the value
+    this.#apiSender.send('extensions-development-folders-changed');
+  }
+
+  async addDevelopmentFolder(path: string): Promise<void> {
+    // check we have extension loader
+    if (!this.#extensionLoader) {
+      throw new Error('No extension loader available');
+    }
+
+    // check the path is not already in the list
+    if (this.#developmentFolders.includes(path)) {
+      throw new Error(`Path ${path} already exist in the list`);
+    }
+
+    // before adding the path, check it's a valid extension path
+    const analyzedExtension = await this.#extensionLoader.analyzeExtension(path, false);
+    // if there is an error, abort
+    if (analyzedExtension.error) {
+      throw new Error(analyzedExtension.error);
+    }
+
+    this.#developmentFolders.push(path);
+
+    // persist the changes
+    await this.saveToConfiguration();
+    this.#onDidUpdateDevelopmentFolders.fire(this.getDevelopmentFolders());
+
+    // start the extension
+    await this.#extensionLoader.loadExtension(analyzedExtension);
+  }
+
+  async removeDevelopmentFolder(path: string): Promise<void> {
+    this.#developmentFolders = this.#developmentFolders.filter(folder => folder !== path);
+
+    // persist the changes
+    await this.saveToConfiguration();
+
+    // dispatch
+    this.#onDidUpdateDevelopmentFolders.fire(this.getDevelopmentFolders());
```

Comment:
should the extension be unloaded here, as it has been loaded in `addDevelopmentFolder`?

Reply:
the UI prevent to stop a running extension.

so the only thing that could be added is a check to throw an error if there is a matching extension running